lec15-data-viz.Rmd

```diff
@@ -1,4 +1,333 @@
+```{r}
+g <- ggplot(iris, aes(x = Sepal.Width, y = Sepal.Length)) 
+g + geom_point()
+```
+
+We can add other "layers" of graphs to our first graph. Let's add a trend line to our scatterplot:
+```{r}
+g + geom_point() + 
+  geom_smooth() 
+```
+
+We can also "facet" our graphs, which means dividing a plot into subplots based on the values of one or more discrete variables.
+```{r}
+g + geom_point() + geom_smooth() + facet_grid(.~Species)
+```
+
+There are lots of ways to organize your facet. See "faceting" at the bottom of [ggplot cheetsheet](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf).
+
+
+Let's say we don't want to facet, but we want to colour the dots depending on the species:
+```{r}
+ggplot(iris, aes(x = Sepal.Width, y = Sepal.Length, color = Species)) + geom_point()
+```
+
+We can change other aesthetic features of the graph. Let's get rid of the color aspect of our structure and change the colour and size of the points:
+```{r}
+ggplot(iris, aes(x = Sepal.Width, y = Sepal.Length)) + geom_point(color="steelblue", size=4)
+```
+
+Going back to our graph coloured by species, what if we want to change other aspects, like the theme, or centre the title?
+
+```{r}
+ggplot(iris, aes(x = Sepal.Width, y = Sepal.Length, color = Species)) + geom_point() +
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  theme_bw() +
+  theme(plot.title = element_text(hjust = 0.5))
+```
+What about changing the font? Let's get rid of our preconfigured theme for a moment, which changes many aspects of the aesthetics at once. Let's spend some time building our own theme...
+
+```{r}
+ggplot(iris, aes(x = Sepal.Width, y = Sepal.Length, color = Species)) + geom_point() +
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  theme(plot.title = element_text(hjust = 0.5), 
+        text = element_text(size = 8, family='Times'),
+        legend.title = element_blank())
+  
+```
+
+
+You can add your own alterations to themes. For example, here is an APA theme built on top of the base theme_bw(). Themes can be saved into variables so they don't need to be typed out every time you use a graph.
+
+
+```{r}
+apatheme <- theme_bw()+
+  theme(panel.grid.major = element_blank(),
+        panel.grid.minor = element_blank(),
+        panel.border = element_blank(),
+        axis.line = element_line(),
+        text = element_text(family = 'Times'),
+        legend.title = element_blank())
+
+ggplot(iris, aes(x = Sepal.Width, y = Sepal.Length, color = Species)) + geom_point() +
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  apatheme()
+```
+
+[http://ggplot2.tidyverse.org/reference/theme.html](This is a resource on which parts of the plot are controlled by different theme parameters when and if you create your own theme.)
+
+
+## Bar graphs
```

Comment:
I would prefer if you included the info from my comment and what you said you are thinking of saying during class in the lecture material. It helps the students when they work on their own and will also be helpful for us in possible future iterations of this class.

The content from my previous comment:

> Could you add a short theory section on when it is appropriate to plot parameters of a distribution (barplot, boxplot), the distribution shape (violinplot), and all the point in the distribution (dot/swarm/hive plot)?

> The general point I would like them to understand is that although many people publish barplots, they are generally a terrible choice when representing the mean + error of a distribution and only really useful to visualize counts or similar. You could relate it to Tufte's philosophy of the ink being used for the data rather than decoration as the rectangles in a bar plot. Some good points are being made in [this poster](biostat.mc.vanderbilt.edu/wiki/pub/Main/TatsukiKoyama/Poster3.pdf). In most situation, it is a great idea to show the individual observations and overlay some distribution parameters in the form of box/violin/mean/median indicators as needed, the obvious exception being when data points oversaturates the plot and this is where 2d histograms/hexbins become useful.